pkg/client/client_cached.go

```diff
@@ -35,6 +36,23 @@ func (c *cacheClientset) ClusterPackages() ClusterPackageInterface {
 	}
 }
 
+func (c *cacheClientset) Packages(ns string) PackageInterface {
+	p := c.PackageV1Alpha1Client.Packages(ns)
+	if c.packageStore == nil {
+		return p
+	}
+	return &readWriteCacheClient[v1alpha1.Package, v1alpha1.PackageList]{
+		p,
+		readOnlyCacheClient[v1alpha1.Package, v1alpha1.PackageList]{
+			p,
+			c.packageStore,
+			func(items []v1alpha1.Package) v1alpha1.PackageList {
+				return v1alpha1.PackageList{Items: items}
+			},
+		},
```

Comment:
This probably doesn't check the namespace, we should probably add this to the `readOnlyCacheClient`

Reply:
Good point, everything still worked but it always went into the fallback because the namespaced items were never found in the cache store. They are stored in this format: `namespace/name`. I tried to make the cache optionally namespace-aware, where the list-part is unused (we never `GetAll` on namespaced packages), but the `Get` case seems to work. Please have a look.